src/components/sectionList/listView/useModelListView.tsx

```diff
@@ -0,0 +1,224 @@
+import { useMemo, useCallback } from 'react'
+import { useQueryClient } from 'react-query'
+import { z } from 'zod'
+import {
+    getViewConfigForSection,
+    sectionNames,
+    useModelSectionHandleOrThrow,
+} from '../../../lib'
+import { useDataStoreValues } from '../../../lib/dataStore'
+import {
+    queryCreators,
+    useMutateDataStoreValues,
+} from '../../../lib/dataStore/useDataStore'
+import { ModelListView } from './types'
+
+const maintenanceNamespace = 'maintenance'
+const configurableColumnsKey = 'modelListViews'
+
+const valuesQueryKey = [
+    queryCreators.getValues({
+        namespace: maintenanceNamespace,
+        key: configurableColumnsKey,
+    }),
+]
+
+const modelListViewSchema = z.object({
+    name: z.string(),
+    sectionModel: z.string(),
+    columns: z.array(z.string()),
+    filters: z.array(z.string()),
+})
+
+type DataModelListView = z.infer<typeof modelListViewSchema>
```

Comment:
I find the name very confusing as we already have the type `ModelListView`. I can see that they're different, but the names are quite confusing as they represent the same data in slightly different formats.

I think this applies to the entire file, there's a lot of complexity involved. Not in the actual code, but in the design of types and code organization. I had to go through some files to understand what a column/filter of the `ModelListView` actually is. Sometimes they're a string, but publicaccess for example is an object with a component, so there are different places that determine which component is being used.

What I'd like to see is that we have some high-level tests (maybe with cypress) so we can address some of the complexity at some point. I think it falls outside of the scope of this PR as we have some time pressure, too

Reply:
I agree, I will try to cleanup this a bit. Would `DataStoreModelListView` be better? This describes the data in the `DataStore` - which is different from the internal format.

Reply:
> Would DataStoreModelListView be better? This describes the data in the DataStore - which is different from the internal format.

Yeah that'd be much clearer already